tests/src/kakarot/instructions/test_environmental_information.py

```diff
@@ -114,20 +113,28 @@ def test_gasprice(self, cairo_run):
             cairo_run("test__exec_gasprice")
 
     class TestExtCodeHash:
-        @SyscallHandler.patch("IERC20.balanceOf", lambda addr, data: [0, 1])
         @SyscallHandler.patch(
-            "IAccount.account_type", lambda addr, data: [int.from_bytes(b"CA", "big")]
+            "IERC20.balanceOf",
+            lambda sn_addr, data: (
+                [0, 1] if sn_addr == EXISTING_ACCOUNT_SN_ADDR else [0, 0]
```

Comment:
isn't is`data[0]  == EXISTING_ACCOUNT` ?

Reply:
The goal is to test with balance=0 and balance=1 for EXISTING_ACCOUNT, otherwise for an non-existing account the balance is _always_ 0